android-core/plugins/org.eclipse.andmore/src/org/eclipse/andmore/internal/build/builders/BaseBuilder.java

```diff
@@ -320,10 +320,16 @@ protected void abortOnBadSetup(@NonNull IJavaProject javaProject,
         if (projectState == null) {
             projectState = Sdk.getProjectState(javaProject.getProject());
         }
-
+        
+        // the project state can still be null, 
+        // e.g. if there is no project.properties file
+        if (projectState == null) {
```

Comment:
Speaking as a complete outsider to the project, and sorry for intruding. 

I like unit tests, but if the project stagnates due to lack of unit tests, it would be a shame.